app/models/friendship.rb

```diff
@@ -0,0 +1,6 @@
+class Friendship < ApplicationRecord
+  belongs_to :confirmer, foreign_key: :confirmer_id, class_name: 'User'
+  belongs_to :requester, foreign_key: :requester_id, class_name: 'User'
```

Comment:
Check the comment I left on your User model. When you implement the changes, this will also have to change.